tensorflow/python/ops/array_ops.py

```diff
@@ -1085,6 +1085,7 @@ def boolean_mask(tensor, mask, name="boolean_mask"):
     tensor:  N-D tensor.
     mask:  K-D boolean tensor, K <= N and K must be known statically.
     name:  A name for this operation (optional).
+    axis: The axis in `tensor` to mask from. Defaults to the first dimension.
```

Comment:
`boolean_mask` already can work as an n-dimensional mask.  Currently if `mask.shape.ndims = K`, it masks over the first `K` dimensions of `tensor`.  So the docstring should be changed to reflect the new behavior if e.g. `axis != None` and `mask` is multi-dimensional.